libs/s3/s3_versioning_common_test_lib.py

```diff
@@ -20,41 +20,343 @@
 #
 """Contains common functions for S3 Versioning tests."""
 import logging
+import random
 
+from commons.exceptions import CTException
 from commons.utils import assert_utils
-from libs.s3 import s3_test_lib
-from libs.s3 import s3_versioning_test_lib
+from commons.utils import s3_utils
+from config.s3 import S3_CFG
+from libs.s3.s3_test_lib import S3TestLib
+from libs.s3.s3_versioning_test_lib import S3VersioningTestLib
 from libs.s3.s3_restapi_test_lib import S3AccountOperationsRestAPI
 
 LOG = logging.getLogger(__name__)
 
 
-def create_s3_user_get_s3lib_object(
-        user_name: str = None,
-        email_id: str = None,
-        password: str = None) -> tuple:
+def create_s3_user_get_s3lib_object(rest_obj: S3AccountOperationsRestAPI,
+                                    user_name: str = None,
+                                    email_id: str = None,
+                                    password: str = None) -> tuple:
     """
     Function will create s3 accounts with specified account name and email-id.
 
+    :param rest_obj: S3AccountOperationsRestAPI object for performing CSM operations.
     :param str user_name: Name of user to be created.
     :param str email_id: Email id for user creation.
     :param password: user password.
-    :return tuple: It returns multiple values such as access_key,
-    secret_key and S3 objects which required to perform further operations.
+    :return: tuple, returns multiple values such as access_key, secret_key and S3 objects
+        which are required to perform further operations.
     """
-    rest_obj = S3AccountOperationsRestAPI()
-    LOG.info("Creating account with name %s and email_id %s",
-             account_name, email_id)
+    LOG.info("Creating user with name %s and email_id %s",
+             user_name, email_id)
     new_user = rest_obj.create_s3_account(user_name=user_name, 
                                           email_id=email_id,
                                           passwd=password)
     assert_utils.assert_true(new_user[0], new_user[1])
     access_key = new_user[1]["access_key"]
     secret_key = new_user[1]["secret_key"]
     del rest_obj
-    LOG.info("Successfully created the S3 account")
-    s3_obj = s3_versioning_test_lib.S3VersioningTestLib(access_key=access_key, 
-                                                        secret_key=secret_key,
-                                                        endpoint_url=S3_CFG["s3_url"])
+    LOG.info("Successfully created the S3 user")
+    s3_obj = S3VersioningTestLib(access_key=access_key, 
+                                 secret_key=secret_key,
+                                 endpoint_url=S3_CFG["s3_url"])
     response = (s3_obj, access_key, secret_key)
     return response
+
+def check_list_object_versions(s3_ver_test_obj: S3VersioningTestLib,
```

Comment:
Do you really like this name S3VersioningTestLib?  FYI that all libs are actually test libs. It would be readable even if you keep the name S3Versioning.

Reply:
Resolving this review comment based on our earlier discussion, as S3VersioningTestLib needs to follow the naming convention laid down in existing S3 libs.